uaclient/entitlements/repo.py

```diff
@@ -204,17 +218,20 @@ def process_contract_deltas(
             "Updating '%s' apt sources list on changed directives.", self.name
         )
 
-        if delta_apt_url:
+        if not self._check_apt_url_is_applied(delta_apt_url):
             orig_entitlement = orig_access.get("entitlement", {})
             old_url = orig_entitlement.get("directives", {}).get("aptURL")
             if old_url:
                 # Remove original aptURL and auth and rewrite
                 repo_filename = self.repo_list_file_tmpl.format(name=self.name)
                 apt.remove_auth_apt_repo(repo_filename, old_url)
-        self.remove_apt_config()
-        self.setup_apt_config()
+
+            self.remove_apt_config()
+            self.setup_apt_config()
+
         if delta_packages:
             self.install_packages(package_list=delta_packages)
+
```

Comment:
Let's also leave a conditional message here if we are updating packages due to deltas
```suggestion
            logging.info msg ("Installing packages on change directives: {}".format(", ".join(delta_packages)) 
            self.install_packages(package_list=delta_packages)

```